### PR TITLE
Exp. UI: Add glance card

### PR DESCRIPTION
**Pull request by c727**
fix: https://github.com/home-assistant/ui-schema/issues/13

```yaml
      - type: glance
        title: Living room
        entities:
          - sensor.temperature
          - sensor.humidity
          - binary_sensor.door
          - binary_sensor.window
          - light.living_room
```
More features, like rename or header picture, can be added later
unbenannt

### CODE REVIEW

**Comment by balloob** [comment]
What is vertical about this? 😄

File: src/panels/experimental-ui/hui-entities-vertical-card.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import computeStateDisplay from '../../common/entity/compute_state_display.js';
+import computeStateName from '../../common/entity/compute_state_name.js';
+
+import '../../components/entity/state-badge.js';
+import '../../components/ha-card.js';
+
+import EventsMixin from '../../mixins/events-mixin.js';
+import LocalizeMixin from '../../mixins/localize-mixin.js';
+
+const VALID_DOMAINS = ['binary_sensor', 'light', 'sensor'];
```
Comment (balloob) [inline src/panels/experimental-ui/hui-entities-vertical-card.js]: I don't think this is up to us. Let's just allow anything and people can see if it doesn't work?

File: src/panels/experimental-ui/hui-entities-vertical-card.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    return config.title;
+  }
+
+  _computeEntities(config) {
+    return config && config.entities && Array.isArray(config.entities) ?
+      config.entities.filter(entity => VALID_DOMAINS.includes(computeDomain(entity))) : [];
+  }
+
+  _showEntity(item, states) {
+    return states && states[item] ? true : false;
```
Comment (balloob) [inline src/panels/experimental-ui/hui-entities-vertical-card.js]: I think we should currently lean to fail visible instead of fail silently

**Comment by balloob** [comment]
What do you think of naming this card "glance" as it allows a user to quickly glance at some info?

For the future PR that contains picture, we should just take in a url via the config. Not based on an entity.

**Comment by raccettura** [comment]
How does this deal with width and how multiple cards show? ie.  If you have several of them with 2 entities would you end up with a tall narrow stack and a lot of empty space on the side? What if you have 3 of them but they have a dozen entities in them? Do they run off the screen?

I really like this, but like the plant card, have found it annoying that they don't utilize screen space terribly well, especially on my wall mounted tablet. IMHO in landscape could/should easy be able to have 2 next to each other with 4-6 entities and avoid some scrolling.

**Comment by balloob** [comment]
@raccettura it would be up to the user to decide how many entities they want to put in the card to fit their needs.

**Comment by c727** [comment]
Yeah the name is an epic fail :D

Currently it starts a new row after 5 entities or if <100px space left

If you want all in one row we can change it?

File: src/panels/experimental-ui/hui-entities-vertical-card.js (hunk 2 above)
Comment (c727) [reply src/panels/experimental-ui/hui-entities-vertical-card.js]: This is only to check if the entity exists in states, but yes, for the things we should show errors

**Comment by raccettura** [comment]
I don't see a problem with that logic, my concerns regarding space usage might be a big more generic. The plant card for example doesn't tile like groups so in a landscape view it just stacks making for lots of scrolling and potentially dead space on the right of the screen no matter how many/few entities you have. IMHO it should tile so that the screen space is fully utilized and scrolling is minimized.

I do see a lot of potential uses for this visually.

**Comment by c727** [comment]
I need some help in CSS
- line space betweens flex rows
- remove this hack in line 46
unbenannt

**Comment by balloob** [comment]
I think the tiling issue can be solved in a future PR.

File: src/panels/experimental-ui/hui-glance-card.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+          display: flex;
+          flex-wrap: wrap;
+        }
+        .entity {
+          padding: 0 4px;
+          display: flex;
+          flex-direction: column;
+          align-items: center;
+          cursor: pointer;
+          width: 18%;
```
Comment (balloob) [inline src/panels/experimental-ui/hui-glance-card.js]: Drop this and throw a `justify-content: space-between` on `.entities`. See also https://css-tricks.com/snippets/css/a-guide-to-flexbox/#article-header-id-6

File: src/panels/experimental-ui/hui-glance-card.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          padding: 4px 0 12px;
+        }
+        .header .name {
+          @apply --paper-font-common-nowrap;
+        }
+        .entities {
+          padding: 4px 0;
+          display: flex;
+          flex-wrap: wrap;
+          flex-basis: 20%;
```
Comment (balloob) [inline src/panels/experimental-ui/hui-glance-card.js]: This still seems very aimed at 5 entities 😉 I'm fine with that for the initial release, but it's my understanding that if we drop this and the `width` from `.entity`, it should look good.

**Comment by balloob** [comment]
Was playing with it and there is no perfect solution. That's why I decided to make the decision to just aim for a width of 5 entities.

**Comment by balloob** [comment]
image

**Pull request merged**